src/Builder/Visitor/AddPluginToChainedPluginCollectionVisitor.php

```diff
@@ -0,0 +1,218 @@
+                }
+
+                $stmt->expr = $this->processMethodCall($stmt->expr);
+            }
+
+            foreach ($node->stmts as $stmt) {
+                if ($this->pluginInserted) {
+                    break;
+                }
+
+                if (!$this->isStatementAddPluginMethodCall($stmt)) {
+                    continue;
+                }
+
+                $arguments = $this->createAddPluginArguments();
+                $newMethodCall = (new BuilderFactory())
+                    ->methodCall($stmt->expr, $stmt->expr->name, $arguments);
+                $stmt->expr = $newMethodCall;
+                $this->pluginInserted = true;
+
+                break;
+            }
+
+            return $node;
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param \PhpParser\Node\Expr\MethodCall $methodCall
+     *
+     * @return \PhpParser\Node\Expr\MethodCall
+     */
+    protected function processMethodCall(MethodCall $methodCall): MethodCall
+    {
+        /** @var \PhpParser\Node\Arg $arg */
+        foreach ($methodCall->args as $arg) {
+            if ($arg->value instanceof New_ && $arg->value->class->toString() === $this->classMetadataTransfer->getBefore()) {
+                $arguments = $this->createAddPluginArguments();
+                $newMethodCall = (new BuilderFactory())
+                    ->methodCall($methodCall->var, $methodCall->name, $arguments);
+                $methodCall->var = $newMethodCall;
+
+                $this->pluginInserted = true;
+
+                return $methodCall;
+            }
+
+            if ($arg->value instanceof New_ && $arg->value->class->toString() === $this->classMetadataTransfer->getAfter()) {
+                $arguments = $this->createAddPluginArguments();
+                $newMethodCall = (new BuilderFactory())
+                    ->methodCall($methodCall, $methodCall->name, $arguments);
+
+                $this->pluginInserted = true;
+
+                return $newMethodCall;
+            }
+        }
+
+        if ($methodCall->var instanceof MethodCall) {
+            $methodCall->var = $this->processMethodCall($methodCall->var);
+        }
+
+        return $methodCall;
+    }
+
+    /**
+     * @param \PhpParser\Node\Stmt $stmt
+     *
+     * @return bool
+     */
+    protected function isStatementAddPluginMethodCall(Stmt $stmt): bool
+    {
+        if ($stmt instanceof Expression === false) {
+            return false;
+        }
+
+        if ($stmt instanceof Expression && $stmt->expr instanceof MethodCall === false) {
+            return false;
+        }
+
+        if (
+            $stmt instanceof Expression
+            && $stmt->expr instanceof MethodCall
+            && strpos(strtolower($stmt->expr->name->toString()), 'add') === false
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @return array<\PhpParser\Node\Arg>
+     */
+    protected function createAddPluginArguments(): array
```

Comment:
Do we use the same functionality [here?](https://github.com/spryker-sdk/integrator/pull/34/files#diff-24afc36bd198ab41f3aa8becd527a01aef99a430dad2ef96eaab2ed2a6529692R148)
Could you please recheck all visitors?